dask_cuda/device_host_file.py

```diff
@@ -1,28 +1,39 @@
 from zict import Buffer, File, Func
 from zict.common import ZictBase
-from distributed.protocol import deserialize_bytes, serialize_bytes
+from distributed.protocol import (
+    deserialize,
+    deserialize_bytes,
+    serialize,
+    serialize_bytes,
+)
 from distributed.worker import weight
 
 from functools import partial
 import os
 
 from .is_device_object import is_device_object
+from .utils import move_frames_to_device, move_frames_to_host
 
 
-def _serialize_if_device(obj):
-    """ Serialize an object if it's a device object """
-    if is_device_object(obj):
-        return serialize_bytes(obj, on_error="raise")
-    else:
-        return obj
+def _serialize(obj):
+    """ Serialize an object and moves frames to host if it's a device object """
+    header, frames = serialize(
+        obj, serializers=["cuda", "dask", "pickle"], on_error="raise"
+    )
+    if header["serializer"] == "cuda":
+        frames = move_frames_to_host(frames)
+    return header, frames
 
 
-def _deserialize_if_device(obj):
-    """ Deserialize an object if it's an instance of bytes """
-    if isinstance(obj, bytes):
-        return deserialize_bytes(obj)
-    else:
+def _deserialize(obj):
+    """ Deserialize an object if it's a serialized object, thus assumes it's a tuple of
+    length 2, and moves frames to device if the serialized object is a CUDA object """
+    if not isinstance(obj, tuple):
```

Comment:
Is there a situation where this isn't true?

Reply:
There was a situation of `obj` being an `int`. I should probably check why is that.